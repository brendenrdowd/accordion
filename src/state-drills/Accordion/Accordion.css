button{
  display:block;
  margin:5px auto;
  padding:2px;
}

li{
  list-style: none;
}

main{
  text-align:center;
  padding:10px 35%;
}
p{
  text-align: left;
}