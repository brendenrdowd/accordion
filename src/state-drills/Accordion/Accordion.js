import React from 'react'
import './Accordion.css'

export default class Accordion extends React.Component {
  // constructor(props) {
  //   super(props);
  // }
  static defaultProps = { sections: [] };

  state = {
    currentSectionIndex: null
  };

  handleButtonClick(index) {
    this.setState({ currentSectionIndex: index })
  }
  // renderButtons() {
  //   return this.props.sections.map((section, index) => (
  //       <button key={index} onClick={() => this.handleButtonClick(index)}>
  //         {section.title}
  //       </button>
  //   ))
  // }

  // renderContent() {
  //   const currentSection = this.props.sections[this.state.currentSectionIndex]
  //   return (
  //     <p className='content'>
  //       {currentSection.content}
  //     </p>
  //   )
  // }

  renderContent(section, index, currentSectionIndex){
    return (
      <li className="item" key={index}>
        <button onClick={() => this.handleButtonClick(index)}>{section.title}</button>
        {(currentSectionIndex === index) && <p>{section.content}</p>}
      </li>
    )
  }

  render() {
    const { currentSectionIndex } = this.state;
    const { sections } = this.props;
    return (
      <ul className="accordion">
        {sections.map((section,index) => 
          this.renderContent(section, index,currentSectionIndex)
        )}
      </ul>
    )
  }
}

