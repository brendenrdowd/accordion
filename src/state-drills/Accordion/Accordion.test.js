import React from 'react';
import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Accordion from './Accordion';

describe(`Accordion Component`, () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'This is the first section. It has a lot to say about being number 1.',
    },
    {
      title: 'Section 2',
      content: 'This is the second section. It has many things to say about the advantages of being a number 2.',
    },
    {
      title: 'Section 3',
      content: 'This is the third section. Hey, still a podium!',
    },
  ]

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the first tab by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('closes the first tab and opens any clicked tab', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})