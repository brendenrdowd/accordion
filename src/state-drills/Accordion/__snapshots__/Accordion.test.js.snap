// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first tab and opens any clicked tab 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    Section 1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Section 2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Section 3
  </button>
  <div
    className="content"
  >
    This is the second section. It has many things to say about the advantages of being a number 2.
  </div>
</div>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `<div />`;

exports[`Accordion Component renders the first tab by default 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    Section 1
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Section 2
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Section 3
  </button>
  <div
    className="content"
  >
    This is the first section. It has a lot to say about being number 1.
  </div>
</div>
`;
